marin/post_training/tpu_attention.py

```diff
@@ -0,0 +1,223 @@
+# adapted from:
+# https://github.com/stanford-crfm/levanter/blob/main/src/levanter/models/attention.py
```

Comment:
Why do we need to copy this from levanter? don't we have a dependency on levanter?  If there are changes, say what they are

Reply:
This is not copied from levanter. The levanter flash attention file (https://github.com/stanford-crfm/levanter/blob/main/src/levanter/models/flash_attention.py) serves more as a reference implementation of flash attention from scratch but this current file uses tpu-optimized kernels (splash_attention_kernel, paged_attention). The use of splash_attention_kernel is from MaxText.